kpi-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java

```diff
@@ -0,0 +1,302 @@
+    String endDate;
+
+    @Option(names = {"-ss", "--skip-save"}, description = "Set this to skip save the KPI as a datasource", required = false)
+    boolean skipSave;
+
+    Logger logger = LoggerFactory.getLogger(KPIQueries.class);
+
+    public class ProfileSchemaDeserializer extends StdDeserializer<ProfileSchema> {
+
+        public ProfileSchemaDeserializer() {
+            this(null);
+        }
+
+        public ProfileSchemaDeserializer(Class<?> vc) {
+            super(vc);
+        }
+
+        @Override
+        public ProfileSchema deserialize(JsonParser jp, DeserializationContext ctxt)
+                throws IOException, JsonProcessingException {
+            JsonNode node = jp.getCodec().readTree(jp);
+
+            List<AttributeSpec> attributes = Arrays.asList(ctxt.readTreeAsValue(node.get("customAttributes"), CustomAttributeSpec[].class));
+            attributes.addAll(Arrays.asList(ctxt.readTreeAsValue(node.get("bucketAttributes"), BucketAttributeSpec[].class)));
+
+            return new ProfileSchema(
+                    node.get("project").asText(),
+                    node.get("name").asText(),
+                    node.has("title") ? node.get("title").asText(null) : null,
+                    node.has("description") ? node.get("description").asText(null) : null,
+                    ctxt.readTreeAsValue(node.get("names"), ProfileNames.class),
+                    ctxt.readTreeAsValue(node.get("primarySource"), DataSourceSelection.class),
+                    Arrays.asList(ctxt.readTreeAsValue(node.get("joins"), JoinSourceSelection[].class)),
+                    node.has("userId") ? node.get("userId").asText(null) : null,
+                    attributes,
+                    Arrays.asList(ctxt.readTreeAsValue(node.get("attributeTags"), AttributeTag[].class))
+            );
+        }
+    }
+
+    protected <T> T getOrDefault(Supplier<T> function, T defaultValue) {
+        try {
+            T value = function.get();
+            if (value == null) {
+                return defaultValue;
+            }
+            return value;
+
+        } catch (NullPointerException | NotFoundException e) {
+            return defaultValue;
+        }
+    }
+
+    @Override
+    public void run() {
+        var sessionExample = new SessionExample();
+        CortexSession cortexSession = sessionExample.getCortexSession();
+        RemoteStorageEnvLocator remoteStorageEnvLocator = new RemoteStorageEnvLocator(mapAsJavaMap(cortexSession.getContext().getSparkSession().conf().getAll()), new LocalRemoteStorageClient(null));
+        String bucketProtocol = remoteStorageEnvLocator.get().getProtocol();
+        String connectionType = remoteStorageEnvLocator.get().getType().toString();
+        Pair<String, String> bucketApiEndpoint = remoteStorageEnvLocator.get().getFeedbackAggregatorConfig();
+        String profilesBucket = remoteStorageEnvLocator.get().getBucketName("cortex-profiles");
```

Comment:
This bit looks interesting. I think this usage of the `RemoteStorageEnvLocater` seems fine, though I'm not sure if it was intended to be directly instantiated..

I would just suggest putting some of this in separate method to make it a bit more obvious what a user would change if they wanted to use something different (plus they won't get hung up on all the details).

For example, I don't know what the `bucketApiEndpoint` fields are or what `getFeedbackAggregatorConfig()` returns, but it looks like you're building a connection to the profiles bucket in the cluster, so that could be its own method (returning the `connectionString`).